apps/qubit/modules/settings/actions/diacriticsAction.class.php

```diff
@@ -0,0 +1,104 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class SettingsDiacriticsAction extends SettingsEditAction
+{
+    // Arrays not allowed in class constants
+    public static $NAMES = [
+        'diacritics',
+    ];
+
+    public function earlyExecute()
+    {
+        parent::earlyExecute();
+
+        $this->updateMessage = $this->i18n->__('Diacritics settings saved.');
+
+        $this->settingDefaults = [
+            'diacritics' => QubitSetting::getByName('diacritics')->getValue(['sourceCulture' => true]),
+        ];
+    }
+
+    public function execute($request)
+    {
+        DefaultEditAction::execute($request);
+
+        if ($request->isMethod('post')) {
+            $this->form->bind($request->getPostParameters(), $request->getFiles());
+
+            if ($this->form->isValid()) {
+                if (0 === intval($request->getPostParameters()['diacritics']) || (1 === intval($request->getPostParameters()['diacritics']) && 0 !== $request->getFiles()['mappings']['size'])) {
+                    $this->processForm($request);
+                    $this->uploadDiacritics($request);
+
+                    $this->getUser()->setFlash('notice', $this->context->i18n->__('Diacritics setting updated. Rebuild the search index for the changes to be applied.'));
+                    $this->redirect(['module' => 'settings', 'action' => 'diacritics']);
+                } else {
+                    $this->getUser()->setFlash('error', $this->context->i18n->__('To enable diacritics setting, please upload a char_filter mapping YAML.'));
+                    $this->redirect(['module' => 'settings', 'action' => 'diacritics']);
+                }
+            }
+        }
+    }
+
+    public function processForm($request)
+    {
+        sfConfig::set('app_diacritics', $request->getPostParameters()['diacritics']);
+        QubitSetting::findAndSave('diacritics', $request->getPostParameters()['diacritics'], ['sourceCulture' => true]);
+    }
+
+    public function uploadDiacritics($request)
+    {
+        $file = $request->getFiles('mappings');
+
+        if (1 === (int) $request->getPostParameters()['diacritics']) {
+            if ('application/x-yaml' === $request->getFiles()['mappings']['type']) {
+                // Move uploaded file to new location to pass off to background arFileImportJob.
+                try {
+                    $file = Qubit::moveUploadFile($file);
+                } catch (sfException $e) {
+                    $this->getUser()->setFlash('error', $e->getMessage());
+                }
+
+                // Replace /config/diacritics_mapping.yml with uploaded file
+                $diacriticsMappingPath = '/atom/src/plugins/arElasticSearchPlugin/config/diacritics_mapping.yml';
```

Comment:
Should this be an absolute path given that where AtoM's code lives may vary? Also I'm wondering if the `config` dir in the root of the AtoM installation might be better given that files in plugin conf directories usually don't get changed.

Reply:
is there a way to get AtoM's base code location instead of using `/atom/src/`? i chose to keep it in `arElasticSearchPlugin/config` since that's where `mapping.yml` is located and I reused some of the code for that in `arElasticSearchPlugin.class.php`, but could change that if you think it's better.

Reply:
Yeah, you might be right about putting it in the plugin directory so I'd say go with that (I'm a bit fuzzy on Symfony's config best practices, tbh).

To get the plugin's config path dynamically you can do:

    sfConfig::get('sf_plugins_dir').DIRECTORY_SEPARATOR.'arElasticSearchPlugin'.DIRECTORY_SEPARATOR.'config';